lib/meta/module.js

```diff
@@ -94,7 +94,16 @@ function generateMeta (pwa) {
 
     // Launch Screen Image (IOS)
     if (options.mobileAppIOS && !find(this.options.head.link, 'rel', 'apple-touch-startup-image')) {
-      this.options.head.link.push({ rel: 'apple-touch-startup-image', href: iconBig.src })
+      this.options.head.link.push({ href: options.iosSplash.iphonese, media: '(device-width: 320px) and (device-height: 568px) and (-webkit-device-pixel-ratio: 2)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.iphone6, media: '(device-width: 375px) and (device-height: 667px) and (-webkit-device-pixel-ratio: 2)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.iphoneplus, media: '(device-width: 621px) and (device-height: 1104px) and (-webkit-device-pixel-ratio: 3)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.iphonex, media: '(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.iphonexr, media: '(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.iphonexsmax, media: '(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.ipad, media: '(device-width: 768px) and (device-height: 1024px) and (-webkit-device-pixel-ratio: 2)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.ipadpro1, media: '(device-width: 834px) and (device-height: 1112px) and (-webkit-device-pixel-ratio: 2)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.ipadpro2, media: '(device-width: 834px) and (device-height: 1194px) and (-webkit-device-pixel-ratio: 2)', rel: 'apple-touch-startup-image' })
+      this.options.head.link.push({ href: options.iosSplash.ipadpro3, media: '(device-width: 1024px) and (device-height: 1366px) and (-webkit-device-pixel-ratio: 2)', rel: 'apple-touch-startup-image' })
```

Comment:
```suggestion
      for (const { href, width, height, pixelRatio } of options.iosSplash) {
        this.options.head.linnk.push({
          href,
          media: `(device-width: ${Math.round(width / pixelRatio)}px)
            and (device-height: ${Math.round(height / pixelRatio)}px)
            and (-webkit-device-pixel-ratio: ${pixelRatio})`,
          rel: 'apple-touch-startup-image'
        })
      }
```

Reply:
Great idea, I didn't realise there was a pixel ratio used for those dimensions!